migrations/kdvh/import.go

```diff
@@ -0,0 +1,324 @@
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"slices"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/rickb777/period"
+
+	"migrate/lard"
+	"migrate/utils"
+)
+
+type ImportConfig struct {
+	Verbose     bool     `short:"v" description:"Increase verbosity level"`
+	BaseDir     string   `short:"p" long:"path" default:"./dumps/kdvh" description:"Location the dumped data will be stored in"`
+	TablesCmd   string   `short:"t" long:"table" default:"" description:"Optional comma separated list of table names. By default all available tables are processed"`
+	StationsCmd string   `short:"s" long:"station" default:"" description:"Optional comma separated list of stations IDs. By default all station IDs are processed"`
+	ElementsCmd string   `short:"e" long:"elemcode" default:"" description:"Optional comma separated list of element codes. By default all element codes are processed"`
+	Sep         string   `long:"sep" default:","  description:"Separator character in the dumped files. Needs to be quoted"`
+	HasHeader   bool     `long:"header" description:"Add this flag if the dumped files have a header row"`
+	Skip        string   `long:"skip" choice:"data" choice:"flags" description:"Skip import of data or flags"`
+	Email       []string `long:"email" description:"Optional email address used to notify if the program crashed"`
+
+	Tables   []string
+	Stations []string
+	Elements []string
+
+	OffsetMap map[StinfoKey]period.Period // Map of offsets used to correct (?) KDVH times for specific parameters
+	StinfoMap map[StinfoKey]StinfoParam   // Map of metadata used to query timeseries ID in LARD
+	KDVHMap   map[KDVHKey]Timespan        // Map of `from_time` and `to_time` for each (table, station, element) triplet. Not present for all parameters
+}
+
+func (config *ImportConfig) setup() {
+	if len(config.Sep) > 1 {
+		slog.Warn("'--sep' only accepts single-byte characters. Defaulting to ','")
+		config.Sep = ","
+	}
+	if config.TablesCmd != "" {
+		config.Tables = strings.Split(config.TablesCmd, ",")
+	}
+	if config.StationsCmd != "" {
+		config.Stations = strings.Split(config.StationsCmd, ",")
+	}
+	if config.ElementsCmd != "" {
+		config.Elements = strings.Split(config.ElementsCmd, ",")
+	}
+	config.CacheMetadata()
+}
+
+func (config *ImportConfig) Execute([]string) error {
+	config.setup()
+
+	// Create connection pool for LARD
+	pool, err := pgxpool.New(context.TODO(), os.Getenv("LARD_STRING"))
+	if err != nil {
+		slog.Error(fmt.Sprint("Could not connect to Lard:", err))
+		return err
+	}
+	defer pool.Close()
+
+	for _, table := range KDVH {
+		if config.Tables != nil && !slices.Contains(config.Tables, table.TableName) {
+			continue
+		}
+		table.Import(pool, config)
+	}
+
+	return nil
+}
+
+func (table *Table) Import(pool *pgxpool.Pool, config *ImportConfig) (rowsInserted int64) {
+	defer utils.SendEmailOnPanic("importTable", config.Email)
+
+	if table.importUntil == 0 {
```

Comment:
I guess? It's similar to matching an Option, except here None is 0 (Go default values :grimacing:). I can extend the comment

Reply:
Reasons to love Go 🙃 

I would personally rather have an extra bool field instead of overloading importUntil, but I don't mind if you just extend the comment to explain this quirk